direnvrc

```diff
@@ -155,4 +167,9 @@ use_devenv() {
     log_status "using cached devenv shell"
     _nix_import_env "$profile_rc"
   fi
+
+  # Watch any extra files configured via `config.direnv.watchFiles` for changes
+  # This picks up new files and reloads the shell on initial run
+  # TODO: deduplicate already watched file paths?
```

Comment:
Could easily be done with assosiative arrays but that needs bash 4. Seeing as macos only ships with version 3, that is probably a no go.

I wonder if it would makes sense to move some of this logic to the rust core where you would have much more freedom and power to do what you want. But keeping everything in simple bash has the benefit that the barrier to enter/debug is quite a bit lower.